src/neighbors.cc

```diff
@@ -1148,22 +1149,33 @@ peerDNSConfigure(const ipcache_addrs *ia, const Dns::LookupDetails &, void *data
 }
 
 static void
-peerRefreshDNS(void *data)
+peerDnsRefreshStart()
 {
-    if (eventFind(peerRefreshDNS, nullptr))
-        eventDelete(peerRefreshDNS, nullptr);
+    for (const auto &p: CurrentCachePeers())
+        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p.get());
 
+    /* Reconfigure the peers every hour */
+    peerScheduleDnsRefreshCheck(3600.0);
+}
+
+static void
+peerDnsRefreshCheck(void *data)
+{
     if (!data && 0 == stat5minClientRequests()) {
         /* no recent client traffic, wait a bit */
-        eventAddIsh("peerRefreshDNS", peerRefreshDNS, nullptr, 180.0, 1);
+        peerScheduleDnsRefreshCheck(180.0);
         return;
     }
 
-    for (const auto &p: CurrentCachePeers())
-        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p.get());
+    peerDnsRefreshStart();
+}
 
-    /* Reconfigure the peers every hour */
-    eventAddIsh("peerRefreshDNS", peerRefreshDNS, nullptr, 3600.0, 1);
+static void
+peerScheduleDnsRefreshCheck(const double delayInSeconds)
```

Comment:
Please try to move this up, where the old eventFind() was, to simplify the diff.